fs/sshfs/sshfs.py

```diff
@@ -155,6 +156,28 @@ def listdir(self, path):  # noqa: D102
         with convert_sshfs_errors('listdir', path):
             return self._sftp.listdir(_path)
 
+    def scandir(self, path, namespaces=None, page=None):  # noqa: D102
+        self.check()
+        _path = self.validatepath(path)
+        _namespaces = namespaces or ()
+        _start, _stop = page or (None, None)
+        try:
+            with convert_sshfs_errors('scandir', path, directory=True):
+                _stat_iter = self._sftp.listdir_iter(_path)
```

Comment:
While I have no idea why, `listdir_iter` is failing the [`test_move_same_fs`](https://github.com/PyFilesystem/pyfilesystem2/blob/master/fs/test.py#L1697). `fs.move` calls `removetree` which calls `walk` which calls `scandir`; the `scandir` call then hangs forever. 

I'd revert back to `listdir_attr` for now, which should already give a sufficient speed boost.

Reply:
I believe the reason is that in the `SFTP.listdir_iter` it assumes that all outstanding async requests on `self` belong to the current generator, but that's not true if multiple generators are in-flight (as would happen during a lazy depth-first search, aka `Walker(search="depth")`.